smithy-cli/src/main/java/software/amazon/smithy/cli/commands/ValidationEventFormatOptions.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.cli.commands;
+
+import java.util.function.Consumer;
+import software.amazon.smithy.cli.ArgumentReceiver;
+import software.amazon.smithy.cli.CliError;
+import software.amazon.smithy.cli.CliPrinter;
+import software.amazon.smithy.cli.HelpPrinter;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.validation.ValidationEvent;
+import software.amazon.smithy.model.validation.ValidationEventFormatter;
+
+final class ValidationEventFormatOptions implements ArgumentReceiver {
+
+    private static final String VALIDATION_FORMAT = "--validation-format";
+
+    enum ValidationFormat {
+        TEXT {
+            @Override
+            void print(CliPrinter printer, ValidationEventFormatter formatter, ValidationEvent event) {
+                printer.println(formatter.format(event));
+            }
+        },
+
+        CSV {
+            @Override
+            void beginPrinting(CliPrinter printer) {
+                printer.println("severity,id,shape,file,message,hint,suppressionReason");
+            }
+
+            @Override
+            void print(CliPrinter printer, ValidationEventFormatter formatter, ValidationEvent event) {
+                printer.println(
+                        String.format("\"%s\",\"%s\",\"%s\",\"%s\",\"%s\",\"%s\",\"%s\"",
+                                      event.getSeverity().toString(),
+                                      formatCsv(event.getId()),
+                                      event.getShapeId().map(ShapeId::toString).orElse(""),
+                                      formatCsv(
+                                              event.getSourceLocation().getFilename()
+                                              + ':' + event.getSourceLocation().getLine()
+                                              + ':' + event.getSourceLocation().getColumn()),
+                                      formatCsv(event.getMessage()),
+                                      formatCsv(event.getHint().orElse("")),
+                                      formatCsv(event.getSuppressionReason().orElse(""))));
+            }
+        };
+
+        void beginPrinting(CliPrinter printer) {}
+
+        abstract void print(CliPrinter printer, ValidationEventFormatter formatter, ValidationEvent event);
+
+        void endPrinting(CliPrinter printer) {}
+
+        private static String formatCsv(String value) {
+            // Replace DQUOTE with DQUOTEDQUOTE and escape newlines.
+            return value.replace("\"", "\"\"").replace("\n", "\\n");
```

Comment:
```suggestion
            return value.replace("\"", "\"\"").replace("\n", "\\n").replace("\r", "\\r");
```

Rationale, if someone uses `String.format` with `%n` that will will add carriage return for some OS's.